samples/machine-learning/half-moons/host.py

```diff
@@ -10,8 +10,6 @@
 plt.style.use('ggplot')
 
 import qsharp
-qsharp.packages.add("Microsoft.Quantum.MachineLearning::0.11.2004.2825")
-qsharp.reload()
```

Comment:
Just wanted to call out that the removal of these lines won't work properly until the upcoming QDK release has completed (which includes the support for reading the packages from the `.csproj`). ETA for that release is August 24. (That said, I think this wouldn't work on the most recent releases _anyway_, since it references an 0.11.* version.)

Reply:
Yes, I thought about it. But since it doesn't work anyways... how bad is to have it unusable until the 24th? 

What it is important is that this is the sample that we are going to have for when the video is released, since this procedure is what I'm going to record.

Reply:
Yes, that makes sense to me. I suspect it's probably fine, but I'll let @cgranade make the call as the owner of these samples.

Reply:
In general, if the sample is broken until release, it's best to either defer merging the PR, or targeting a release branch. I agree it's not the end of the world given its five days until the sample is fixed, but we do want to make sure people have a good first impression of samples since that's a primary entry point to the Quantum Development Kit.

Reply:
That makes sense, @cgranade. It looks like the sample may already be broken in its current state (since it's referencing an old 0.11.* version of the package), so perhaps updating that version would be a good immediate step for this PR?

@geduardo, I can take the action item to come update these samples for the new functionality after the next release is complete.

Reply:
Now that we're on the other side of the release, I think we can resume this PR and use the new automatic package loading functionality introduced in the August. Thoughts, @geduardo and @rmshaffer?

Reply:
True, I almost forgot about this PR. What new functionality was added? 

Relevant information: A video showing how to run these samples is being produced. Changes in the process to run the samples should be avoided, or if it needs to be changed it should be done ASAP.

Reply:
Automatic package loading was added in the previous release, so that the extraneous "fail" messages at the console can be avoided by including a csproj file with the relevant packages in the same folder as the Q# notebook.

```
<Project Sdk="Microsoft.Quantum.Sdk/0.12.20082513">
    <PropertyGroup>
        <OutputType>Library</OutputType>
        <TargetFramework>netstandard2.1</TargetFramework>
        <IQSharpLoadAutomatically>true</IQSharpLoadAutomatically>
    </PropertyGroup>
    <ItemGroup>
        <PackageReference Include="Microsoft.Quantum.MachineLearning" Version="0.12.20082513" />
    </ItemGroup>
</Project>
```

(example adapted from https://docs.microsoft.com/quantum/user-guide/host-programs?tabs=tabid-python#using-q-code-from-other-projects-or-packages-1)

Given that this would significantly improve how the sample looks, I'd suggest getting it in ASAP if you and @KittyYeungQ agree.

Reply:
@geduardo @cgranade I have the changes for automatic package loading available in PR https://github.com/microsoft/Quantum/pull/411. I'd be happy to merge that one; I believe it's ready.

Reply:
Reviewing now, then. Thanks!

Reply:
@rmshaffer  @cgranade Let me get this clear. How does this automatic package loading affect this PR since using the Conda environment packages were already loaded correctly?  If I understand it correctly, we still need the Conda environment since we are loading Python packages like Numpy or Matplotlib.

Reply:
@geduardo Yes, sorry, "package" is overloaded here. 😊 This "automatic package loading" feature is not for the Python packages, but for the NuGet packages like `Microsoft.Quantum.MachineLearning`. You can see the relevant changes in PR #411. The only difference from what you have here is that it also adds `<IQSharpLoadAutomatically>true</IQSharpLoadAutomatically>` to the .csproj files.

Reply:
Ok, I see, then I think the video doesn't need to be modified @KittyYeungQ. Once one of the two PRs gets merged, the other one will get some small conflicts I think. You can merge #411 and then I'll resolve the conflicts here.

Reply:
I was wrong! No conflicts since both implement the same changes in the "conflicting" files 😆 Thanks!